### ISSUE TITLE
System config file causes tests to fail

### ISSUE THREAD

**Issue by nathanhaigh** [hpcugent/easybuild-framework#1509]
I've been trying to get EasyBuild setup and running on my Ubuntu 14.04 server. I've noticed that test fail if I have a `/etc/easybuild.d/config.cfg` file specifying that I'm using Lmod as the modules tool.

Setup the environment:

```bash
$ export TEST_EASYBUILD_MODULES_TOOL=Lmod
$ EB_PREFIX=${HOME}/.local/easybuild
$ module use "${EB_PREFIX}/modules/all"
$ module load EasyBuild
```

Run tests without a `/etc/easybuild.d/config.cfg` file:

```bash
$ python -m test.framework.config
...........
----------------------------------------------------------------------
Ran 11 tests in 22.972s

OK
```

```bash
$ python -m test.framework.options
..................Skipping test_from_pr, no GitHub token available?
.Skipping test_from_pr, no GitHub token available?
...............................
----------------------------------------------------------------------
Ran 50 tests in 387.709s

OK
```

Create a config file specifying we are using the Lmod modules tool:

```bash
$ cat << EOF | sudo tee /etc/easybuild.d/config.cfg
[config]
modules-tool = Lmod
EOF
```

Rerun the tests:

```bash
$ python -m test.framework.config
F..........
======================================================================
FAIL: test_XDG_CONFIG_env_vars (__main__.EasyBuildConfigTest)
Test effect of XDG_CONFIG* environment variables on default configuration.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/nhaigh/.local/easybuild/software/EasyBuild/2.4.0/lib/python2.7/site-packages/easybuild_framework-2.4.0-py2.7.egg/test/framework/config.py", line 433, in test_XDG_CONFIG_env_vars
    self.assertEqual(eb_go.options.configfiles, cfg_files)
AssertionError: Lists differ: ['/etc/easybuild.d/config.cfg'... != ['/tmp/eb-QToJF5/eb-K78MTs/hom...

First differing element 0:
/etc/easybuild.d/config.cfg
/tmp/eb-QToJF5/eb-K78MTs/homedir/.config/easybuild/config.cfg

First list contains 1 additional elements.
First extra element 1:
/tmp/eb-QToJF5/eb-K78MTs/homedir/.config/easybuild/config.cfg

- ['/etc/easybuild.d/config.cfg',
-  '/tmp/eb-QToJF5/eb-K78MTs/homedir/.config/easybuild/config.cfg']
? ^

+ ['/tmp/eb-QToJF5/eb-K78MTs/homedir/.config/easybuild/config.cfg']
? ^


----------------------------------------------------------------------
Ran 11 tests in 24.218s

FAILED (failures=1)
```

```bash
$ python -m test.framework.options
..................Skipping test_from_pr, no GitHub token available?
.Skipping test_from_pr, no GitHub token available?
.F.............................
======================================================================
FAIL: test_generate_cmd_line (__main__.CommandLineOptionsTest)
Test for generate_cmd_line.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/nhaigh/.local/easybuild/software/EasyBuild/2.4.0/lib/python2.7/site-packages/easybuild_framework-2.4.0-py2.7.egg/test/framework/options.py", line 1674, in test_generate_cmd_line
    self.assertEqual(ebopts.generate_cmd_line(), [])
AssertionError: Lists differ: ['--modules-tool=Lmod'] != []

First list contains 1 additional elements.
First extra element 0:
--modules-tool=Lmod

- ['--modules-tool=Lmod']
+ []

----------------------------------------------------------------------
Ran 50 tests in 396.083s

FAILED (failures=1)

```

Here's info about my setup:

```bash
$ python -V
Python 2.7.6
$ type module
module is a function
module () 
{ 
    eval $($LMOD_CMD bash "$@");
    [ $? = 0 ] && eval $(${LMOD_SETTARG_CMD:-:} -s sh)
}
$ type -f module
bash: type: module: not found
$ module --version

Modules based on Lua: Version 6.0.1  2015-06-10 16:45
    by Robert McLay mclay@tacc.utexas.edu

$ module av EasyBuild

------------------------------------------------------------------------------------ /home/nhaigh/.local/easybuild/modules/all -------------------------------------------------------------------------------------
   EasyBuild/2.4.0

Use "module spider" to find all possible modules.
Use "module keyword key1 key2 ..." to search for all possible modules matching any of the "keys".

$ which -a eb
/home/nhaigh/.local/easybuild/software/EasyBuild/2.4.0/bin/eb
$ eb --version
This is EasyBuild 2.4.0 (framework: 2.4.0, easyblocks: 2.4.0) on host acpfg-bioserv2.
```

**Comment by nathanhaigh** [comment]
I should have looked at existing issues. I would have realised this is already reported in bug #1467. However, this might provide some additional info.

**Comment by boegel** [comment]
@nathanhaigh: this is indeed a duplicate of #1467, so let's close this and follow up in #1467 (please drop a comment in there to tell us you're suffering from this too)

It's just an issue with the tests not being isolated from a system-wide configuration file, there's nothing wrong with EasyBuild itself.

To run the tests with Lmod to check your installation, using `export TEST_EASYBUILD_MODULES_TOOL=Lmod` is indeed the right way, see also http://easybuild.readthedocs.org/en/latest/Unit-tests.html#modules-tool-to-use-for-running-tests

**Comment by boegel** [comment]
fixed in #1513, @nathanhaigh: please verify?
